.github/workflows/test.yml

```diff
@@ -0,0 +1,146 @@
+# This is a GitHub workflow defining a set of jobs with a set of steps.
+# ref: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions
+#
+name: Test
+
+on:
+  pull_request:
+    branches: ["**"]
+  push:
+    branches: [main, master]
```

Comment:
Is this restriction needed? Are there branches that are created but don't need CI testing?

Reply:
If someone forks this repo, it will create jobs in the forks when pushing to the PR branches which isn't needed. I think it's quite nice to clearly signal the intent.

Reply:
I retained the restriction and added a comment, let me know if you think we should remove it. I think it makes sense though as it conserves compute resources and avoids potential clutter of notifications of failed CI jobs that can show up twice etc otherwise.

Reply:
Personally, I really *like* being able to run tests on my forks before opening PRs. I'm not asking for a change, just noting that I personally benefit from lacking this config rather often.

Reply:
I'd also like to run tests on my own branches. See related discussions:
- https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/323/files#r155242906
- https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/388
- https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/500

Reply:
I think it is a bit wasteful of GitHub's free compute to make this the default behavior even though it would be nice sometimes to have, I would prefer it myself as the default behavior but it feels like a compromise worth making to not be wasteful of the free compute resources given to us.

Create a suggestion and merge it if you want though, I'm not strongly opined about this.